### PR TITLE
Fix/issue 36610 always call api

### PR DESCRIPTION
**Pull request by kmbcook**
### Details

ProfilePage now calls openPublicProfilePage regardless of value of hasMinimumDetails.

### Fixed Issues

$ https://github.com/Expensify/App/issues/36610
PROPOSAL: https://github.com/Expensify/App/issues/36610#issuecomment-1949528548

### Tests

Same as QA steps.

- [X] Verify that no errors appear in the JS console

### Offline tests

This issue is only relevant when online.

### QA Steps

1. Go to a room chat.
2. Click on the user avatar to open profile page.
3. Check console messages or network inspector to verify that openPublicProfilePage API call has been made.

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/14853493/fae0bdfe-3461-43c2-a5f7-d760dda3c259

https://github.com/Expensify/App/assets/14853493/92223ae1-28f9-4dc6-b5cb-7f7bd26079f2

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/14853493/d5e35454-c9da-4c49-af9a-09ebcd19e93a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/14853493/64b8f56e-d069-49bb-b6d2-effef3ebfc09

https://github.com/Expensify/App/assets/14853493/7ccb4cbe-1304-4b2e-bfce-1adc5ab1b298

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/14853493/6df0c7d7-a27c-420a-a706-bc50cba22adf

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/14853493/a3d475ef-80f7-4d7a-8ba4-652874f92cc6

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/14853493/7ecb9111-0520-40b5-a8f6-232fc0171fd5

</details>

### CODE REVIEW

**Comment by kmbcook** [comment]
Oops, dragging and dropping my videos didn't work. I'll edit above to correct. Also, two of them are too big I just noticed, so I'll have to take care of that.

**Comment by kmbcook** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by cubuspl42** [comment]
Weird, I don't get why the CLA check is failing.

File: src/pages/ProfilePage.js [hunk 1]
```diff
@@ -136,12 +136,9 @@ function ProfilePage(props) {
     const shouldShowNotificationPreference = !_.isEmpty(props.report) && props.report.notificationPreference !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
     const notificationPreference = shouldShowNotificationPreference ? props.translate(`notificationPreferencesPage.notificationPreferences.${props.report.notificationPreference}`) : '';
 
-    // eslint-disable-next-line rulesdir/prefer-early-return
     useEffect(() => {
-        if (ValidationUtils.isValidAccountRoute(accountID) && !hasMinimumDetails) {
```
Comment (cubuspl42) [inline src/pages/ProfilePage.js:141]: What are your thought on removing the `isValidAccountRoute(accountID)` check? I know we said "unconditionally" multiple times, but I'm not sure how literal was that. I thought it was related to `hasMinimumDetails` check in practice.

Do you know what account routes are "invalid"?

**Comment by kmbcook** [comment]
> Weird, I don't get why the CLA check is failing.

Is it because I only signed my last commit, and not the others?

File: src/pages/ProfilePage.js (hunk 1 above)
Comment (kmbcook) [reply src/pages/ProfilePage.js:141]: Here is the new PR
https://github.com/Expensify/App/pull/37223

Comment (cubuspl42) [reply src/pages/ProfilePage.js:141]: CLA signing is unrelated to Git GPG commit signing. The only common part is the word "sign". CLA is an agreement that you accept as a human being. Git commit signing is a cryptographic proof of your authorship of Git commits.

Anyway, as the CLA check is successful on the second PR, let's move the discussion to https://github.com/Expensify/App/pull/37223

Please close **this** PR!

Comment (cubuspl42) [reply src/pages/ProfilePage.js:141]: Also, applying changes on the PR is a very standard procedure and, in this project, it's done by adding new commits on top of the old ones. Sometimes people choose commit names as trivial as "Update after PR feedback", but in this case something like "Add the `isValidAccountRoute` check back" would be more suitable.

In other project, force-pushing to create a new "PR revision" could be preferred, but in Expensify force-pushing is disallowed (or at least discouraged) after the first review round. I didn't make this rule.

This is just FYI, as for now, we're moving to https://github.com/Expensify/App/pull/37223.

Comment (kmbcook) [reply src/pages/ProfilePage.js:141]: Got it, thanks for explaining. I'll close this PR.

**Pull request closed without merging**